Assignment_1.py

```diff
@@ -0,0 +1,233 @@
+# %%
+import pandas as pd
+import numpy as np
+
+from nilearn import plotting
+import nibabel as nib
+
+from scipy import signal, stats
+
+import matplotlib.pyplot as plt
+from matplotlib.ticker import FixedLocator, FixedFormatter
+
+import warnings
+# %%
+
+def standardize_data(bold_data, chunks):
+
+    """Standardizes and detrends the data.
+    
+    Parameters
+    ----------
+    bold_data : array_like 
+        4-D array with time dimension on last index The input data
+    
+    chunks_df : array_like
+        array containing the indicies of chenks ordered by their time
+
+    Returns
+    -------
+    temp : ndarray
+        z-score of the input data.
+    
+    """
+    for chunk in chunks:
+        bold_data[:,:,:,chunk] = \
+            stats.zscore(\
+            signal.detrend(\
+                bold_data[:,:,:,chunk],type='linear'))
+    
+    return(bold_data)
+
+
+def create_selection_mask(bold, x:int=None,y:int=None,z:int=None, r:int=None):
+    """Create mask for the BOLD image based on user input.
+
+    if any of the values are non-int, no mask will be created.
+    
+    Parameters
+    ----------
+    bold: array_like
+        target data for masking. This variable is used to get the dimensions.
+
+    x: int, optional
+        x dimension
+
+    y: int, optional
+        y dimension
+
+    z: int, optional
+        z dimension
+
+    r: int, optional
+        radius of selection.
+
+    Returns
+    -------
+    ret : array_like
+        returns a list of indicies if validation is true
+    """
+    
+    if any(list(map(lambda m: m is None, [x,y,z]))) :
+        warnings.warn('inputs contain None. No selection applied')
+        return(np.ones(bold.shape[:3])) # no mask
+
+    else:
+
+        if r is not None:
+
+            # creating distance matrix
+            i,j,k = np.indices(bold.shape[:3], sparse=True)
+            distmat = np.sqrt((i-x)**2 + (j-y)**2 + (k-z)**2)
+
+            # creating the mask based on distance
+            mask_cube =np.zeros(bold.shape[:3])
+            mask_cube[distmat<= r] = 1
+            return(mask_cube == 1)
+
+        else:
+
+            return(np.ones(bold.shape[:3])) # no mask
+
+
+def plot_MRI(image,bold,mask,title=None, ax=None):
+    """gets the input and produces searchlight mask if applicable.
+    
+    Parameters
+    ----------
+    images: int
+    index of the desired image
+    bold: nibabel.nifti1.Nifti1Image
+        BOLD image data
+    mask: array_like
+        Mask that is created to cover unnecessary data
+        
+    Returns
+    -------
+        MRI plots
+    """
+
+
+def RSA_RDM_producer(bold_array_masked, labeled_df):
+    """Produces RDM and Model RDM and plots them.
```

Comment:
It's generally a good idea to have a function do only one thing. In this case having a function to create the RDM and a separate function to plot it.